src/main/scala/lms/transformation/distributed_tensor/distributedTensorUnary.scala

```diff
@@ -45,6 +69,33 @@ trait FixedSizeDistributedTensorUnaryTypeLess extends FixedSizeDistributedTensor
         (() => {
           Accumulate(gradMap(a), Negate(gradMap(s), anno), anno); ()
         }) +=: backwardNodes
+     
+    case Node(s, "tensor_invert", tt::Backend.Const(anno:Anno)::(a:Backend.Sym)::_, _) =>
+        implicit val pos = Adapter.oldSourceMap(s)
+        forwardNodes += node
+
+        (() => {
+          val a_tensor = new TENSOR(transform(a))
+          val square = Mul(a_tensor, a_tensor, anno)
+          val grad = Div(gradMap(a), square, anno)
```

Comment:
should this be `gradMap(s)` instead of `gradMap(a)`?
also there seems to be missing a negation?
if y = x^-1, then dy/dx = -1x^-2

Another thing to take into account is that this implementation will likely contain 3 CUDA kernels (the mul, the div, the neg). It would be interesting if we can use 1 CUDA kernel that does all three things. It can be done either via a later transformation/optimization pass, or by defining a `Invert_grad` kernel that will later lower to a CUDA kernel that does this three things.

Reply:
Done.